### PR TITLE
Latest broken links update

### PR DESCRIPTION
**Pull request by SafeEHA**
Updated 4 of 5 GitHub broken links to Permalinks for #328 
- https://github.com/metal3-io/baremetal-operator/blob/main/pkg/bmc/access.go
- https://github.com/metal3-io/metal3-docs/blob/main/design/capm3-remediation-controller-proposal.md
- https://github.com/metal3-io/baremetal-operator/blob/main/deploy/crds/metal3.io_baremetalhosts_crd.yaml
- https://github.com/metal3-io/metal3-docs/blob/main/design/baremetal-operator/re-introspection-interface.md
- https://cluster-api.sigs.k8s.io/tasks/healthcheck.html

### CODE REVIEW

File: design/baremetal-operator/implicit-boot-mode.md [hunk 1]
```diff
@@ -106,7 +106,7 @@ that is also easy to document.
 The proposal, therefore, is to always use BIOS mode when using IPMI
 and to prefer UEFI when using Redfish. We can implement this by adding
 a new method to the [AccessDetails
-interface](https://github.com/metal3-io/baremetal-operator/blob/main/pkg/bmc/access.go#L27)
+interface](https://github.com/metal3-io/baremetal-operator/blob/dabe5e14bafa00db6ccb37f1169c74ee3dac4425/pkg/hardwareutils/bmc/access.go)
```
Comment (tuminoid) [inline design/baremetal-operator/implicit-boot-mode.md:109]: Please keep the line reference, update to point to correct line as needed.

File: design/baremetal-operator/implicit-boot-mode.md [hunk 2]
```diff
@@ -141,7 +141,7 @@ values later for other hardware platforms.
 ## Design Details
 
 - Update the [AccessDetails
-  interface](https://github.com/metal3-io/baremetal-operator/blob/main/pkg/bmc/access.go#L27)
+  interface](https://github.com/metal3-io/baremetal-operator/blob/dabe5e14bafa00db6ccb37f1169c74ee3dac4425/pkg/hardwareutils/bmc/access.go)
```
Comment (tuminoid) [inline design/baremetal-operator/implicit-boot-mode.md:144]: Please keep the line reference, update to point to correct line as needed.

Comment (SafeEHA) [reply design/baremetal-operator/implicit-boot-mode.md:144]: Ok, would fix that now

File: design/baremetal-operator/implicit-boot-mode.md [hunk 3]
```diff
@@ -106,7 +106,7 @@ that is also easy to document.
 The proposal, therefore, is to always use BIOS mode when using IPMI
 and to prefer UEFI when using Redfish. We can implement this by adding
 a new method to the [AccessDetails
-interface](https://github.com/metal3-io/baremetal-operator/blob/dabe5e14bafa00db6ccb37f1169c74ee3dac4425/pkg/hardwareutils/bmc/access.go)
+interface](https://github.com/metal3-io/baremetal-operator/blob/dabe5e14bafa00db6ccb37f1169c74ee3dac4425/pkg/hardwareutils/bmc/access.go#L27)
```
Comment (tuminoid) [inline design/baremetal-operator/implicit-boot-mode.md:109]: AccessDetails interface is starting at L47.

File: design/baremetal-operator/implicit-boot-mode.md (hunk 2 above)
Comment (tuminoid) [reply design/baremetal-operator/implicit-boot-mode.md]: It is now pointing to wrong line, correct line is L47.

**Pull request closed without merging**